### DISCUSSION TITLE
Opening a dialog from a custom panic hook

### DISCUSSION THREAD

**Discussion by Futsch1** [slint-ui/slint#955] [Q&A]
In my application, I'd like to show a popup to the user in case a panic occurs. I tried to show/run a dialog from the panic hook using this function:
```
pub fn show_error_dialog(error_str: SharedString) {
    slint::slint! {
        import { StandardButton, VerticalBox } from "std-widgets.slint";

        ErrorDialog := Dialog {
            property <string> error_text: "Error";
            VerticalBox {
                alignment: start;
                Text {
                    text: error_text;
                }
                StandardButton { kind: close; }
            }
        }
    }
    let error_dialog = ErrorDialog::new();
    error_dialog.set_error_text(error_str);
    error_dialog.run();
}
```

However, the window is not shown properly and the dialog does not wait for the button to be pressed. How would this work?

**Comment by tronical** [comment]
Ohh, this could be an issue with a nested event loop and the way that we're using winit. Do you know by chance if you're using the GL backend or Qt?

**Reply by Futsch1** [reply]
I am using the GL backend.
